remove legacyIndicatorMenu

aeschli opened the pull request:
Fixes https://github.com/microsoft/vscode/issues/175930

aeschli committed:
remove legacyIndicatorMenu

joyceerhl reviewed (changes requested):
vscode.dev still uses this and there are a few other consumers in the wild.

aeschli committed:
Remove Menu.StatusBarWindowIndicatorMenu

aeschli committed:
Merge branch 'main' into aeschli/faint-primate

aeschli commented:
@joyceerhl Please have a look. I now removed `StatusBarWindowIndicatorMenu` and replaced its usages to `StatusBarRemoteIndicatorMenu`
In the web API, we still use the term `window indicator` even if we call it RemoteIndicator in our code. IMO that's ok, and I don't want start a swirl. There are two more web APIs also use that term.
For the download command to use `StatusBarRemoteIndicatorMenu` I added a new group `general`. These commands will be grouped at the end of the remote menu.

joyceerhl reviewed (commented):
I think the commands from vscode.dev need to be updated to have a group, otherwise they'll get silently dropped from the remote menu with this change.

aeschli commented:
no plans to implement
